### DISCUSSION TITLE
Brew Doctor and Linking, and In-App Software Updater

### DISCUSSION THREAD

**Discussion by yordis** [Homebrew/discussions#5637] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.24
ORIGIN: https://github.com/Homebrew/brew
HEAD: 916044581862c32fc2365e8e9ff0b1507a98925e
Last commit: 4 days ago
Core tap JSON: 26 Sep 17:33 UTC
Core cask tap JSON: 26 Sep 17:33 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code --wait
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.46.2 => /usr/local/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  zed
  gnupg
```

### Description of issue

How do I deal with the situation where some apps get updated using the In-App Software Updater? 

`brew Doctor` asked me to link it. Ideally, Brew would play more in a friendly manner with the In-App Software Updater.

**Comment by p-linnane** [comment, marked as answer]
> `brew Doctor` asked me to link it. Ideally, Brew would play more in a friendly manner with the In-App Software Updater.

That's not what that message is saying. It's saying you have 2 things installed that are not correctly linked.

`brew` only knows about the things it does. If you want to use in-app updates, then you should expect `brew` will not know about them. If you want to update via Homebrew, you can run `brew upgrade --greedy` to upgrade casks that have in-app updaters.

**Reply by yordis** [reply]
I was not aware of `--greedy` flag, thank you so much!
